File: vibemynight/frontend/flutter_app/lib/models/facility.dart
class Facility {
  final int id;
  final String name;
  final String? icon;
  final String? description;
  final String status;

  const Facility({
    required this.id,
    required this.name,
    this.icon,
    this.description,
    required this.status,
  });

  factory Facility.fromJson(Map<String, dynamic> json) => Facility(
        id: (json['id'] as num?)?.toInt() ?? 0,
        name: json['name']?.toString() ?? '',
        icon: json['icon']?.toString(),
        description: json['description']?.toString(),
        status: json['status']?.toString() ?? 'ACTIVE',
      );
}

File: vibemynight/backend/spring_boot/src/main/java/com/vibemynight/backend/entity/Facility.java
package com.vibemynight.backend.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EqualsAndHashCode(callSuper = false)
@Entity
@Table(name = "facilities")
public class Facility extends BaseEntity {

    @Column(nullable = false, unique = true)
    private String name;

    /** Icon identifier/name (e.g. an icon-font key or asset name) resolved by the frontend. */
    private String icon;

    @Column(length = 1000)
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private ActiveStatus status = ActiveStatus.ACTIVE;
}
